imgur-screenshot.sh

```diff
@@ -23,6 +23,11 @@ function is_mac() {
   uname | grep -q "Darwin"
 }
 
+
+function is_gnome() {
+  which gnome-screenshot | grep -q "gnome-screenshot"
```

Comment:
There should be a better way to determine if GNOME is currently in use. `gnome-screenshot` might be installed despite GNOME not being used.

PS: There's no need to use `grep` because `which` returns an exit code > 0 when the program was not found

Reply:
A better function name would be "has_gnome_screenshot". Because it's not trying to determine what the current DE is. I don't know about KDE, but gnome-screenshot works in both gnome and unity, and way better than scrot.  I could see adding in other capture methods like maim or ksnapshot.  Is there particular direction you'd like to go with that kind of logic, or would the function rename suffice?

Reply:
I guess it works in Unity because Unity is just a replacement for the GNOME shell, but still uses the GNOME DE. For KDE there is `ksnapshot`.

Perhaps we could use [`${XDG_CURRENT_DESKTOP}`](https://askubuntu.com/a/227669) / `${GDMSESSION}` / `${DESKTOP_SESSION}` to figure out a good default screenshot command (if we're not on macOS X), and fall back to `scrot` if the default is not available.